src/server/server_family.cc

```diff
@@ -3758,8 +3758,10 @@ void ServerFamily::ReplTakeOver(CmdArgList args, const CommandContext& cmd_cntx)
   }
 
   std::error_code res = replica_->TakeOver(ArgS(args, 0), save_flag);
-  if (res)
-    return builder->SendError("Couldn't execute takeover");
+  if (res) {
+    LOG(ERROR) << "Takeover failed with error: " << res << " - " << res.message();
```

Comment:
I suspect you do not need both LOG(ERROR) and ` builder->SendError` as we log errors in main_service. At least reduce the severity to WARNING